pl_bolts/models/detection/yolo/loss.py

```diff
@@ -0,0 +1,302 @@
+            checks that the length of the second dimension matches this number.
+        dtype: Floating-point data type to be used for the one-hot targets.
+
+    Returns:
+        An ``[M, C]`` matrix of target class probabilities.
+    """
+    if targets.ndim == 1:
+        # The data may contain a different number of classes than what the model predicts. In case a label is
+        # greater than the number of predicted classes, it will be mapped to the last class.
+        last_class = torch.tensor(num_classes - 1, device=targets.device)
+        targets = torch.min(targets, last_class)
+        targets = torch.nn.functional.one_hot(targets, num_classes)
+    elif targets.shape[-1] != num_classes:
+        raise ValueError(
+            f"The number of classes in the data ({targets.shape[-1]}) doesn't match the number of classes "
+            f"predicted by the model ({num_classes})."
+        )
+    return targets.to(dtype=dtype)
+
+
+@dataclass
+class Losses:
+    overlap: Tensor
+    confidence: Tensor
+    classification: Tensor
+
+
+class LossFunction:
+    """A class for calculating the YOLO losses from predictions and targets.
+
+    Args:
+        overlap_func: A function for calculating the pairwise overlaps between two sets of boxes. Either a string or a
+            function that returns a matrix of pairwise overlaps. Valid string values are "iou", "giou", "diou", and
+            "ciou" (default).
+        predict_overlap: Balance between binary confidence targets and predicting the overlap. 0.0 means that target
+            confidence is one if there's an object, and 1.0 means that the target confidence is the output of
+            ``overlap_func``.
+        overlap_loss_multiplier: Overlap loss will be scaled by this value.
+        confidence_loss_multiplier: Confidence loss will be scaled by this value.
+        class_loss_multiplier: Classification loss will be scaled by this value.
+    """
+
+    def __init__(
+        self,
+        overlap_func: Union[str, Callable] = "ciou",
+        predict_overlap: Optional[float] = None,
+        overlap_multiplier: float = 5.0,
+        confidence_multiplier: float = 1.0,
+        class_multiplier: float = 1.0,
+    ):
+        overlap_loss_func = None
+        if overlap_func == "iou":
+            overlap_func = box_iou
+        elif overlap_func == "giou":
+            overlap_func = generalized_box_iou
+            overlap_loss_func = generalized_box_iou_loss
+        elif overlap_func == "diou":
+            overlap_func = distance_box_iou
+            overlap_loss_func = distance_box_iou_loss
+        elif overlap_func == "ciou":
+            overlap_func = complete_box_iou
+            overlap_loss_func = complete_box_iou_loss
+
+        if not callable(overlap_func):
+            raise ValueError(
+                f"Unsupported overlap function '{overlap_func}'. Try upgrading Torcvision or using another IoU "
```

Comment:
Small typo 
```suggestion
                f"Unsupported overlap function '{overlap_func}'. Try upgrading Torchvision or using another IoU "
```

Reply:
We're hitting this error in GPU CI `ValueError: Unsupported overlap function 'None'. Try upgrading Torcvision or using another IoU algorithm.`

The versions of `torch` and `torchvision` used in the CI are:
* `torch==1.11.0+cu113`
* `torchvision==0.12.0+cu113`

Reply:
Now the tests use giou loss, which is available in torchvision 0.12. I also improved the error message.